tango_simlib/examples/override_class.py

```diff
@@ -490,3 +496,423 @@ def test_action_stoprainstorm(self, model, tango_dev=None, data_input=None):
+        """The Dish is tracking the commanded pointing positions within the
+        specified SCAN pointing accuracy. (TBC)
+        NOTE: This pointing state is currently proposed and there are
+        currently no requirements for this functionality.
+
+        data_input: str
+            [Timestamp]
+        """
+        raise NotImplementedError("The command handler for the command Scan"
+                                  " is not implemented.")
+
+    def action_setmaintenancemode(self, model, tango_dev=None, data_input=None):
+        """This command triggers the Dish to transition to the MAINTENANCE
+        Dish Element Mode, and returns to the caller. To go into a state that
+        is safe to approach the Dish by a maintainer, and to enable the
+        Engineering interface to allow direct access to low level control and
+        monitoring by engineers and maintainers. This mode will also enable
+        engineers and maintainers to upgrade SW and FW. Dish also enters
+        this mode when an emergency stop button is pressed.
+
+        data_input: None
+        """
+        _allowed_modes = ('STANDBY-LP', 'STANDBY-FP')
+        dish_mode_quant = model.sim_quantities['dishMode']
+        current_mode_enum_val = dish_mode_quant.last_val
+        current_mode_str_val = (
+            dish_mode_quant.meta['enum_labels'][int(current_mode_enum_val)])
+        if current_mode_str_val in _allowed_modes:
+            set_mode = dish_mode_quant.meta['enum_labels'].index('MAINTENANCE')
+            model.sim_quantities['dishMode'].set_val(set_mode, model.time_func())
+            MODULE_LOGGER.info("Dish transition to the OPERATE Dish Element mode.")
+        else:
+            raise DishSimError("Dish is not in {} mode.".format(_allowed_modes))
+
+    def action_setoperatemode(self, model, tango_dev=None, data_input=None):
+        """This command triggers the Dish to transition to the OPERATE Dish
+        Element Mode, and returns to the caller. This mode fulfils the main
+        purpose of the Dish, which is to point to designated directions while
+        capturing data and transmitting it to CSP.
+
+        data_input: None
+        """
+        _allowed_modes = ('STANDBY-FP')
+        dish_mode_quant = model.sim_quantities['dishMode']
+        current_mode_enum_val = dish_mode_quant.last_val
+        current_mode_str_val = (
+            dish_mode_quant.meta['enum_labels'][int(current_mode_enum_val)])
+        if current_mode_str_val in _allowed_modes:
+            set_mode = dish_mode_quant.meta['enum_labels'].index('OPERATE')
+            model.sim_quantities['dishMode'].set_val(set_mode, model.time_func())
+            MODULE_LOGGER.info("Dish transition to the OPERATE Dish Element Mode.")
+        else:
+            raise DishSimError("Dish is not in {} mode.".format(_allowed_modes))
+
+    def action_pntmodelpars(self, model, tango_dev=None, data_input=None):
```

Comment:
Done. Converted the 'SetPntModelPars' command into a writable attribute named 'pointModelPars'.